src/app/modules/main/chat_section/chat_content/input_area/controller.nim

```diff
@@ -229,11 +234,16 @@ proc loadLinkPreviews*(self: Controller, urls: seq[string]) =
     self.messageService.asyncUnfurlUrls(urls)
 
 proc setLinkPreviewEnabled*(self: Controller, enabled: bool) =
-  if(enabled):
+  if enabled and self.settingsService.saveUrlUnfurlingMode(UrlUnfurlingMode.Enabled):
     self.linkPreviewPersistentSetting = UrlUnfurlingMode.Enabled
     self.linkPreviewCurrentMessageSetting = UrlUnfurlingMode.Enabled
-  else:
+  elif not enabled and self.settingsService.saveUrlUnfurlingMode(UrlUnfurlingMode.Disabled):
     self.linkPreviewPersistentSetting = UrlUnfurlingMode.Disabled
     self.linkPreviewCurrentMessageSetting = UrlUnfurlingMode.Disabled
 
   self.delegate.setAskToEnableLinkPreview(false)
+
+proc onUnfurlingModeChanged(self: Controller, value: UrlUnfurlingMode) =
+  self.linkPreviewPersistentSetting = value
+  self.resetLinkPreviews()
```

Comment:
Hmmm, I don't get it 🤔 
`resetLinkPreviews` will set local setting to `AlwaysAsk`:
https://github.com/status-im/status-desktop/blob/f2a20631d60808c2c4e56db878d572d4680b9974/src/app/modules/main/chat_section/chat_content/input_area/controller.nim#L116

Though initially you sync those variables values:
https://github.com/status-im/status-desktop/blob/f2a20631d60808c2c4e56db878d572d4680b9974/src/app/modules/main/chat_section/chat_content/input_area/controller.nim#L56-L57

Reply:
ohh, you're right! On reset is should pick the persisted value.
Fixed